### ISSUE TITLE
Bug: ignoreCase: false setting does not work

### ISSUE THREAD

**Issue by difosfor** [azat-io/eslint-plugin-perfectionist#424]
### Describe the bug

When specifying `settings.perfectionist.ignoreCase: false` I would expect things like union types to be sorted alphabetically. I.e: With upper case before lower case characters. But that does not seem to be the case.

### Code example

With config:
```js
import perfectionist from 'eslint-plugin-perfectionist';

export default {
  plugins: { perfectionist },
  rules: {
    'perfectionist/sort-union-types': 'error',
  },
  settings: {
    perfectionist: { ignoreCase: false },
  },
};
```

```ts
// I'd expect this to be correct:
type IgnoreCaseFalse = 'fooBarge' | 'fooWarg' | 'foobArgs';
// But instead this is:
type IgnoreCaseTrue = 'fooBarge' | 'foobArgs' | 'fooWarg';
```

### ESLint version

v9.11.1

### ESLint Plugin Perfectionist version

v4.3.0

### Additional comments

I prefer to respect case so as to group members by the words of their camelCase names. See example above.

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by hugop95** [comment]
Hello @difosfor 

Unfortunately, this is not exactly a bug, and more of a (very common) misunderstanding regarding this option and how it works (we need to work on that, you're not the first to report it):

- `ignoreCase: true` simply lowercases any input before entering the sorting phase.
- `type: alphabetical` uses `localeCompare`: and the sorted alphabet for `localeCompare` is `aAbBcCdDeE...`, so lowercased or not, `b/B` will be before `w/W`.

> With upper case before lower case characters

You can use a custom sorting type (`type: custom`). Here is an example configuration that should give you what you need: 

```ts
const alphabet = Alphabet
    .generateRecommendedAlphabet()
    .sortByLocaleCompare("en-US")
    .placeAllWithCaseBeforeAllWithOtherCase("uppercase")
    .getCharacters();

export default {
  plugins: { perfectionist },
  rules: {
    'perfectionist/sort-union-types': 'error',
  },
  settings: {
    perfectionist: { 
	  ignoreCase: false,
	  alphabet: alphabet,
	},
  },
};
```

**Comment by difosfor** [comment]
Ok, I'm actually using natural ordering in my complete config. So in that case I guess I should use the following settings:

```ts
{
  plugins: { perfectionist },
  rules: {
    ...pluginRulesError(perfectionist, 'perfectionist'),
    // opt out of a few rules here
  },
  settings: {
    perfectionist: {
      alphabet: Alphabet.generateRecommendedAlphabet()
        .sortByNaturalSort('en-US')
        .placeAllWithCaseBeforeAllWithOtherCase('uppercase')
        .getCharacters(),
      ignoreCase: false,
      type: 'custom',
    },
  },
}
```

Looks like the settings could perhaps be improved indeed.

But this seems to be working alright, so thanks for your help!

PS: I'm using the following function to get all plugin rules from perfectionist and a few other plugins:

```ts
function pluginRulesError(plugin, name) {
  return Object.fromEntries(
    Object.keys(plugin.rules).map((rule) => [`${name}/${rule}`, 'error']),
  );
}
```

**Comment by difosfor** [comment]
I guess this is working as intended.

**Comment by difosfor** [comment]
One more question if you don't mind me picking your brain 😅 

Is it also possible to do this and have non-word characters precede word characters? P.e: `foo | fooBar` instead of `fooBar | foo`.

**Comment by hugop95** [comment]
@difosfor 

> Is it also possible to do this and have non-word characters precede word characters? P.e: foo | fooBar instead of fooBar | foo.

We currently don't have a setting to modify that in Alphabet, but it's probably worth having, I'll study this.

**Comment by azat-io** [comment]
@hugop95 Is it critical bug? Should I create release now?

**Comment by hugop95** [comment]
@azat-io 

Not a critical bug in my opinion:
- Doesn't break compilation or runtime
- Affects ordering of very specific cases
- Only affects `custom` sort type, which is a new feature => doesn't impact the majority of users.

**Comment by difosfor** [comment]
Ah, thank you very much for fixing this guys! Looking forward to the next release 👍

**Comment by azat-io** [comment]
Fixed in 1649bea46f69ee81eac54f73521e821996f86c91.
Released in [v4.5.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v4.5.0).
